pkg/util/template.go

```diff
@@ -214,6 +220,38 @@ func configMapEntriesGenerator(
 	}
 }
 
+// lookup does a dot-separated path lookup on the input struct or map. If the input or any of
+// its children on the targeted path are not a map or struct, it returns nil.
+func lookup(path string, input interface{}) interface{} {
+	obj := reflect.ValueOf(input)
+
+	components := strings.Split(path, ".")
+
+	for i := 0; i < len(components); {
+		component := components[i]
+
+		switch obj.Kind() {
+		case reflect.Struct:
+			obj = obj.FieldByName(component)
+			i++
+		case reflect.Map:
+			obj = obj.MapIndex(reflect.ValueOf(component))
+			i++
+		case reflect.Ptr, reflect.Interface:
+			// Get the thing being pointed to or interfaced, don't advance index
```

Comment:
I'm under the impression that we may have to check `reflect.Value.IsNil` before we dereference the pointer, otherwise the code could panic.

Reply:
Add a check here.